prow/cmd/deck/job_history.go

```diff
@@ -89,22 +91,26 @@ type jobHistoryTemplate struct {
 	Builds       []buildData
 }
 
-func (bucket gcsBucket) readObject(key string) ([]byte, error) {
-	obj := bucket.Object(key)
-	rc, err := obj.NewReader(context.Background())
+func (bucket gcsBucket) readObject(ctx context.Context, key string) ([]byte, error) {
+	rc, err := bucket.Opener.Reader(ctx, fmt.Sprintf("%s://%s/%s", bucket.storageProvider, bucket.name, key))
 	if err != nil {
-		return []byte{}, fmt.Errorf("failed to get reader for GCS object: %v", err)
+		return nil, fmt.Errorf("creating reader for object %s: %w", key, err)
 	}
+	defer rc.Close()
 	return ioutil.ReadAll(rc)
 }
 
 func (bucket gcsBucket) getName() string {
 	return bucket.name
 }
 
-func readLatestBuild(bucket storageBucket, root string) (int64, error) {
+func (bucket gcsBucket) getStorageProvider() string {
```

Comment:
Nit: This func seems superfluous? A private getter that just returns a private property?

Reply:
I did it this way because otherwise I would have to hand over the struct instead of the interface in `getPRBuildData`